framework/src/transfers/MultiAppInterpolationTransfer.C

```diff
@@ -66,492 +83,498 @@ MultiAppInterpolationTransfer::MultiAppInterpolationTransfer(const InputParamete
 }
 
 void
-MultiAppInterpolationTransfer::execute()
+MultiAppInterpolationTransfer::subdomainIDsNode(MooseMesh & mesh,
+                                                Node & node,
+                                                std::set<subdomain_id_type> & subdomainids)
 {
-  _console << "Beginning InterpolationTransfer " << name() << std::endl;
-
-  switch (_current_direction)
-  {
-    case TO_MULTIAPP:
-    {
-      FEProblemBase & from_problem = _multi_app->problemBase();
-      MooseVariableFEBase & from_var = from_problem.getVariable(
-          0, _from_var_name, Moose::VarKindType::VAR_ANY, Moose::VarFieldType::VAR_FIELD_STANDARD);
+  // We need this map to figure out to which subdomains a given mesh point is attached
+  auto & node_to_elem = mesh.nodeToElemMap();
+  auto node_to_elem_pair = node_to_elem.find(node.id());
 
-      MeshBase * from_mesh = NULL;
+  if (node_to_elem_pair == node_to_elem.end())
+    mooseError("Can not find elements for node ", node.id());
 
-      if (_displaced_source_mesh && from_problem.getDisplacedProblem())
-        from_mesh = &from_problem.getDisplacedProblem()->mesh().getMesh();
-      else
-        from_mesh = &from_problem.mesh().getMesh();
+  subdomainids.clear();
+  // Add all subdomain IDs that are attached to node
+  for (auto element : node_to_elem_pair->second)
+  {
+    auto & elem = mesh.getMesh().elem_ref(element);
+    auto subdomain = elem.subdomain_id();
 
-      SystemBase & from_system_base = from_var.sys();
-      System & from_sys = from_system_base.system();
+    subdomainids.insert(subdomain);
+  }
+}
 
-      unsigned int from_sys_num = from_sys.number();
-      unsigned int from_var_num = from_sys.variable_number(from_var.name());
+void
+MultiAppInterpolationTransfer::fillSourceInterpolationPoints(
+    FEProblemBase & from_problem,
+    MooseVariableFEBase & from_var,
+    Point & from_app_position,
+    std::unique_ptr<InverseDistanceInterpolation<LIBMESH_DIM>> & idi)
+{
+  MeshBase * from_mesh = NULL;
+  MooseMesh * from_moose_mesh = NULL;
 
-      auto & fe_type = from_sys.variable_type(from_var_num);
-      bool from_is_constant = fe_type.order == CONSTANT;
-      bool from_is_nodal = fe_type.family == LAGRANGE;
+  // Get libmesh mesh and moose mesh
+  if (_displaced_source_mesh && from_problem.getDisplacedProblem())
+  {
+    from_mesh = &from_problem.getDisplacedProblem()->mesh().getMesh();
+    from_moose_mesh = &from_problem.getDisplacedProblem()->mesh();
+  }
+  else
+  {
+    from_mesh = &from_problem.mesh().getMesh();
+    from_moose_mesh = &from_problem.mesh();
+  }
 
-      if (fe_type.order > FIRST && !from_is_nodal)
-        mooseError("We don't currently support second order or higher elemental variable ");
+  // Moose system
+  SystemBase & from_system_base = from_var.sys();
+  // libmesh system
+  System & from_sys = from_system_base.system();
```

Comment:
Anywhere I see "from", I want to know if we can use `const`. Conceptually I would hope so

Reply:
This question applies to all later "from" code where you are assigning to reference

Reply:
I know on the phone you said you tried to generalize this for both "from" and "to". Maybe that means you can't do `const`

Reply:
`const`, haha

Reply:
I will try to make const and see what will happen